internal/supervisor/import.go

```diff
@@ -0,0 +1,408 @@
+	ID          string
+}
+
+type importProvisioner interface {
+	ExecClusterInstallationCLI(cluster *model.Cluster, clusterInstallation *model.ClusterInstallation, args ...string) ([]byte, error)
+}
+
+type mmctl struct {
+	cluster             *model.Cluster
+	clusterInstallation *model.ClusterInstallation
+	installation        *model.Installation
+
+	*ImportSupervisor
+}
+
+func (m *mmctl) Run(args ...string) ([]byte, error) {
+	args = append([]string{"mmctl", "--format", "json", "--local"}, args...)
+	return m.provisioner.ExecClusterInstallationCLI(m.cluster, m.clusterInstallation, args...)
+}
+
+type team struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type jobResponse struct {
+	ID       string `json:"id"`
+	Status   string `json:"status"`
+	Progress int    `json:"progress"`
+
+	jobResponseData `json:"data"`
+}
+
+type jobResponseData struct {
+	Error      string `json:"error"`
+	LineNumber int    `json:"line_number"`
+	ImportFile string `json:"import_file"`
+}
+
+// NewImportSupervisor creates a new Import Supervisor
+func NewImportSupervisor(awsClient toolsAWS.AWS, awat AWATClient, store installationStore, provisioner importProvisioner, logger logrus.FieldLogger) *ImportSupervisor {
+	return &ImportSupervisor{
+		awsClient:   awsClient,
+		awatClient:  awat,
+		store:       store,
+		logger:      logger,
+		provisioner: provisioner,
+
+		// TODO replace this with the Pod ID from env var
+		ID: model.NewID(),
+	}
+}
+
+// Do checks to see if there is an Import that is ready to be
+// imported, and if so, does that. Otherwise, it does nothing.
+func (is *ImportSupervisor) Do() error {
+	work, err := is.awatClient.GetTranslationReadyToImport(
+		&awat.ImportWorkRequest{
+			ProvisionerID: is.ID,
+		})
+	if err != nil {
+		return errors.Wrap(err, "failed to get a ready Import from the AWAT")
+	}
+	if work == nil {
+		// nothing to do
+		return nil
+	}
+	defer func() {
+		err = is.awatClient.ReleaseLockOnImport(work.ID)
+		if err != nil {
+			is.logger.WithError(err).Warnf("failed to release lock on Import %s", work.ID)
+		}
+	}()
+
+	err = is.importTranslation(work)
+	if err != nil {
+		is.logger.WithError(err).Errorf("failed to perform work on Import %s", work.ID)
+	}
+
+	errorString := ""
+	if err != nil {
+		errorString = err.Error()
+	}
+
+	cerr := is.awatClient.CompleteImport(
+		&awat.ImportCompletedWorkRequest{
+			ID:         work.ID,
+			CompleteAt: time.Now().UnixNano() / int64(time.Millisecond),
+			Error:      errorString,
+		})
+	if cerr != nil {
+		is.logger.WithError(cerr).Errorf("failed to notify AWAT that Import %s is complete", work.ID)
+	}
+
+	return err
+}
+
+// Shutdown is called when the ImportSupervisor is stopped.
+// TODO change Shutdown from a no-op to allowing it to unlock ongoing
+// Imports so that they are detected as interrupted by the AWAT and
+// can be re-served to another Provisioner instance as a cleaner retry
+// pattern
+func (is *ImportSupervisor) Shutdown() {
```

Comment:
Small style pointer. All of the other supervisors use `s` for their methods: `(s *InstallationSupervisor)`

Seems a little cleaner to me, but 0/5

Reply:
Part of it seems to be my brain reading `is` as the literal word and I am getting scary ruby syntax vibes. 😂